ee/control/control_test.go

```diff
@@ -0,0 +1,163 @@
+package control
+
+import (
+	"io"
+	"testing"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type (
+	mockConsumer struct {
+		updateFn func()
+	}
+	mockSubscriber struct {
+		pingFn func()
+	}
+)
+
+func (mc *mockConsumer) Update(io.Reader) {
+	if mc.updateFn != nil {
+		mc.updateFn()
+	}
+}
+func (mc *mockSubscriber) Ping() {
+	if mc.pingFn != nil {
+		mc.pingFn()
+	}
+}
+
+type nopDataProvider struct{}
+
+func (dp nopDataProvider) Get(subsystem, cachedETag string) (etag string, data io.Reader, err error) {
+	return "", nil, nil
+}
+
+func testCS() *ControlService {
+	data := nopDataProvider{}
+	cs := &ControlService{
+		logger:          log.NewNopLogger(),
+		requestInterval: 60 * time.Second,
+		consumers:       make(map[string]consumer),
+		subscribers:     make(map[string][]subscriber),
+		data:            data,
+	}
+
+	return cs
+}
+
+func TestControlServiceRegisterConsumer(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name      string
+		subsystem string
+		c         consumer
+	}{
+		{
+			name:      "empty subsystem",
+			subsystem: "",
+			c:         &mockConsumer{},
+		},
+		{
+			name:      "happy path",
+			subsystem: "desktop",
+			c:         &mockConsumer{},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			cs := testCS()
+			err := cs.RegisterConsumer(tt.subsystem, tt.c)
+			require.NoError(t, err)
```

Comment:
Should this test that the consume is actually registered? I might say it's a subset of the `TestControlServiceUpdate` path.